### PR TITLE
Fix RadialGradientBrush GradientOrigin property usage.

### PR DESCRIPTION
**Pull request by olliholliday**
## What does the pull request do?
The IGradientBrush object includes a GradientOrigin parameter which is used in Direct2D but not Skia - the two engines produce wildly different graphical results when GradientOrigin is set.

This PR attempts to change the Skia implementation to take into account the GradientOrigin property so that they produce the same output.

## What is the current behavior?
The current Skia implementation of GradientBrush ignores the GradientOrigin property and there are no tests that use it which would highlight this.

## What is the updated/expected behavior with this PR?
This PR alters the implementation of GradientBrush for Skia and adds new tests that demonstrate that the Direct2D and Skia implementations are now the same.

## How was the solution implemented (if it's not obvious)?
The implementation of the Skia GradientBrush was changed from using Skia GradientBrush to using Skia TwoPointConicalBrush which matches the behaviour of the Direct2d RadialBrush where a start and end gradient point are configurable. 

One difference between the Skia TwoPointConicGradient brush and the Direct2d GradientBrush is that in the Skia implementation, pixels that fall outside of the gradient map are transparent when GradientOrigin falls outside of the circle described by Center/Radius - whereas in Direct2D the 'background colour' of the whole surface is set to the outer gradient colour. To resolve this, the IRadialGradient implementation has been changed to compose with a background fill colour.

## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [x] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

The picture linked below shows the results before adding the fix in the two windows on the left hand side, and after adding the fix on the right hand side. This shows that before the fix the Skia renderer ignores the GradientOrigin property, and after the fix it contributes to the gradient. https://i.imgur.com/cv8Cvgl.png

With the fix, both sets of results look the same

### CODE REVIEW

File: src/Skia/Avalonia.Skia/DrawingContextImpl.cs [hunk 1]
```diff
@@ -583,13 +583,13 @@ private void ConfigureGradientBrush(ref PaintWrapper paintWrapper, Size targetSi
...
 
-                    // TODO: There is no SetAlpha in SkiaSharp
-                    //paint.setAlpha(128);
+                    var origin = radialGradient.GradientOrigin.ToPixels(targetSize).ToSKPoint();
 
                     // would be nice to cache these shaders possibly?
-                    using (var shader =
-                        SKShader.CreateRadialGradient(center, radius, stopColors, stopOffsets, tileMode))
-                    {
+                    using (var shader = SKShader.CreateCompose(
```
Comment (kekekeks) [inline src/Skia/Avalonia.Skia/DrawingContextImpl.cs:589]: 1) LINQ usage in hot code path
2) we should probably still use CreateRadialGradient when gradient origin is not set

Comment (olliholliday) [reply src/Skia/Avalonia.Skia/DrawingContextImpl.cs]: I've removed the Linq from the PR, though it is still used in this method on lines above. I'm happy to fix that if you prefer.

I've added a comparison to use the normal RadialGradient renderer as it was before, when the origin is the same as the center.

I also notice an issue with the reference point of the stops not being reversed so I've fixed that and added a new test.

**Comment by olliholliday** [comment]
I've noticed a further difference between D2D and Skia in the existing code when gradient stop offsets fall outside of the range 0..1. The Skia implementation clamps the values to that range, whereas the D2D implementation doesn't. 

This is not due to this PR so I don't think it should block. Example here: https://imgur.com/a/DAkVCUp

**Review by grokys — COMMENTED** [review]
LGTM, and appreciate the added render tests, though I must add that I've not been involved much in the rendering side of things for the last few years, so may not be the best person to review.

I guess that one of the questions is: do we want to follow D2D and not Skia here, and I suspect the answer will be "yes, if it's closer to WPF/UWP". I assume WPF/UWP have the same behavior as D2D?

@Gillibald @wieslawsoltes any feedback?

**Comment by Gillibald** [comment]
Looks good to me. We want to follow the WPF API here and this just fixes the current implementation so nothing changes API wise.

**Review by Gillibald — APPROVED** [review]

**Pull request merged**